### PR TITLE
CQ-105332 – Kiosk mode reduces touchable area of content

### PR DESCRIPTION
**Pull request by basecode**
Problem: The Plugin adds an OverlayView on top of the StatusBar in order to prevent user interactions. Although that's generally a good approach, the OverlayView also prevents the user from interacting with content because the OverlayView is enabled all the time and not just when the StatusBar is visible.

Fix: The Cordova app is most likely in immersive fullscreen mode. We can make use of that and install an `OnSystemUiVisibilityChangeListener`. As soon as the SystemUI is triggered we add the OverlayView that prevents user interactions. After ~2 seconds the System allows us to safely remove the SystemUI again and with that the OverlayView.

### CODE REVIEW

**Comment by tripodsan** [comment]
does this also prevent hardware buttons interaction?

**Review by ramboz — APPROVED** [review]
Looks ok.

**Comment by basecode** [comment]
@tripodsan The Plugin does. My fix does not. I updated the description in order to clarify what the fix is actually about.

File: src/android/jk/cordova/plugin/kiosk/KioskActivity.java [hunk 1]
```diff
@@ -78,12 +82,13 @@ protected void onActivityResult(int requestCode, int resultCode,  Intent data) {
...
             if (Settings.canDrawOverlays(this)) {
-                addOverlay();
+                allowImmersiveSystemUiOverlay = true;
             }
         }
     }
+
     //http://stackoverflow.com/questions/25284233/prevent-status-bar-for-appearing-android-modified?answertab=active#tab-top
-    public void addOverlay() {
+    private void addOverlay() {
```
Comment (kptdobe) [inline src/android/jk/cordova/plugin/kiosk/KioskActivity.java:91]: This is an API change. Any bad consequences ?

**Review by kptdobe — APPROVED** [review]

File: src/android/jk/cordova/plugin/kiosk/KioskActivity.java (hunk 1 above)
Comment (visiongeist) [reply src/android/jk/cordova/plugin/kiosk/KioskActivity.java:91]: not if the API is not exposed via JS.

**Review by visiongeist — APPROVED** [review]

File: src/android/jk/cordova/plugin/kiosk/KioskActivity.java (hunk 1 above)
Comment (basecode) [reply src/android/jk/cordova/plugin/kiosk/KioskActivity.java:91]: @kptdobe That's why I changed it so it's clear that this method is actually only used "privately". There was no reason to expose it in the first place to either other Activities or via JS.

**Pull request merged**